_pytest/compat.py

```diff
@@ -82,7 +82,10 @@ def num_mock_patch_args(function):
     return len(patchings)
 
 
-def getfuncargnames(function, startindex=None):
+def getfuncargnames(function, startindex=None, cls=None):
+    if startindex is None and cls is not None:
```

Comment:
I see thanks. But from my understanding this is quite a refactor on how fixture names are obtained and out of scope of this PR. 

This PR attempts to fix fixture support for static methods and doesn't add complexity to the current system IMHO. If you disagree we should close this PR, which I'm OK with.